extensions/2.0/Khronos/KHR_lights_environment/README.md

```diff
@@ -0,0 +1,261 @@
+
+This extension can be used on it's own - ie a glTF asset with only environment map data - for a usecase where the environmental light needs to be distributed.  
+It may also be used together with model data, for usecases where a model shall be displayed in a controlled environment.  
+
+Many 3D tools and engines support image-based global illumination but the exact technique and data formats employed vary.  
+Using this extension, tools can export and engines can import image-based lights and the result should be highly consistent.   
+
+This extension specifies exactly one way to format and reference the environment map to be used, the goals of this are two-fold.  
+First, it makes implementing support for this extension easier.  
+Secondly, it ensures that rendering of the image-based lighting is consistent across runtimes.
+
+A conforming implementation of this extension must be able to load the environment data and render the PBR materials using this lighting.  
+
+Cubemap environment light images are declared as an array of images in the extension, this extension is then placed in the glTF root.  
+These images shall be supplied using a KTX V2 file containing at least one cubemap.  
+
+### Pre Filtering Of Cubemaps
+
+The environment light is defined by a cubemap, this is to simplify realtime implementations as these are likely to use cubemap texture format.  
+Cubemaps shall be supplied without pre-filtered mip-maps for roughness values > 0, pre-filtering shall be done by the client.  
+[See Specular radiance cubemaps](#specular-radiance-cubemaps)  
+
+
+## Declaring An Environment Light
+
+The KHR_lights_environment extension defines an array of environment light cubemaps at the root of the glTF, each scene can reference one these lights.    
+Each environment light definition consists a single cubemap that describes the specular radiance of the scene, the l=2 spherical harmonics coefficients for irradiance, luminance factor and bounding box for localized cubemap.  
+
+The cubemap is defined as an integer reference to one of the texture sources declared in the this extension. 
+This texture source shall contain a cubemap at the layer defined by the texture.   
+
+When this extension is used, images shall use `image/ktx2` as mimeType.  
+The texture type of the KTX v2 file shall be 'Cubemap'  
+
+The following will declare one cubemap to be referenced by a scene.  
+
+
+```json  
+
+"asset": {
+"version": "2.0"
+},
+"extensionsUsed": [
+    "KHR_lights_environment"
+],
+
+"extensions": {
+    "KHR_lights_environment" : {
+        "cubemaps": [
+            {
+                "source": 0
+                "layer": 0
+            }
+        ],
+        "images": [
+            {
+                "name": "environment cubemap 0",
+                "uri": "cubemap0.ktx2",
+                "mimeType": "image/ktx2"
+            }
+        ],  
```

Comment:
I'm curious, why create a new `images` array under the extension rather than using the root `images` array? Similarly, it might be more idiomatic to use the root `textures` array rather than adding `cubemaps` here. For example:

```jsonc
// root
{
  "textures": [{
    "source": 0, 
    "extensions": {"KHR_lights_environment":{"layer": 0}} // optional
  }],
  "images": [{
    "uri": "cubemap.ktx2",
    "mimeType": "image/ktx2"
  }],
  "extensions": [{
    "lights": [{
      "name": "SceneEnvironment",
      "cubemap": 0,
      "irradianceCoefficients": [...],
      ...
    }]
  }],
  ...
}
```

Reply:
Hi @donmccurdy  and thanks for your comment

I agree that it is not the obvious choice to put the cubemaps inside the light extension.  
The reason for this is backwards compatibility and separation.  

By backwards compatibility I mean that since we introduce a new fileformat (ktx) and new texture formats this may break current implementations if they go through the main `images` and/or `textures` arrays and treat them like they are jpg/png.  
This is a behavior which I think must be allowed with the glTF 2.0 spec - there simply is no other formats for images/textures.  

If the cubemaps are included in the root images/textures arrays - how are implementations that does not support this extension supposed to handle those resources?  

By separation I mean that from a data packaging perspective it may make sense to put the cubemaps inside the extension to make it abundantly clear that these resources are only ever used if the extension is supported (and used)

I understand that the behavior to load the cubemaps will be slightly different than to _normal_ resources - but I think it's worth it due to the above reasons.   

Please let me know what you think!

Reply:
IMO, I don't think this would break any existing implementation in the same way that the KHR_texture_basisu extension didn't by adding image/ktx2 mime type in the images. Otherwise an implementation that didn't support KHR_texture_basisu should run into the same issue you described. 

In practice I think most implementation would load what they support and either ignore the rest or load them generically.

For example, in my image loading implementation (WebGPU/WebGL based), for jpg/png or any web loadable image type I load it using an Image element and use createImageBitmap to send it to the GPU. For everything else I simply load it as an array buffer, working under the assumption that any spec compliant object that references that image will know what to do it with it from there. As is the case with KHR_texture_basisu, and now this KHR_lights_environment would as well. In my case with the current spec of this extension, I reused the same Image loading logic I just had to call it again from the extension specific logic rather during the root load logic.

Reply:
Aaah, now I see what you mean.

That would mean that the expected behavior at load time is to check what textures are used and then proceed to load image resources based on that usage.

I will move the images from being included _in_ the extension to be at root level images array instead.